c10/util/string_view.h

```diff
@@ -598,6 +598,22 @@ constexpr inline void swap(
 }
 using string_view = basic_string_view<char>;
 
+// NOTE: In C++20, this function should be replaced by str.starts_with
+constexpr bool string_view_starts_with(
+    std::string_view str,
+    std::string_view prefix) noexcept {
+  return (prefix.size() > str.size()) ? false
+                                      : prefix == str.substr(0, prefix.size());
+}
+
+// NOTE: In C++20, this function should be replaced by str.ends_with
+constexpr bool string_view_ends_with(
+    std::string_view str,
+    std::string_view suffix) noexcept {
+  return (suffix.size() > str.size())
+      ? false
+      : suffix == (str.substr(str.size() - suffix.size(), suffix.size()));
```

Comment:
```suggestion
    return str.size() >= suffix.size() && str.substr(str.size() - suffix.size()) == suffix;
```
Just apply demorgan's law, no need for a ternary operator here.